core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -982,6 +982,7 @@ private Object readResolve() {
         if (jdks == null) {
             jdks = new ArrayList<>();
         }
+        slaveAgentPort = SystemProperties.getInteger(Jenkins.class.getName()+".slaveAgentPort", slaveAgentPort);
```

Comment:
> Most cases this port is fixed because of firewall setup so you don't want anyone to change the value through UI once started.

During debug (that issues constantly appears with security and firewalls) you may want to change it without restarting master. Inconsistency could be checked and reported to administrative monitor.

@daniel-beck i made own hacks for this port by shipping piece of `config.xml` in tests that applies to stock docker-container. 

PS. @kohsuke so would you use PRs next time?